iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -650,4 +650,13 @@ TEST_F(ServiceDiscovery_test, FindServiceReturnsContainerOverflowErrorWhenMoreTh
     EXPECT_TRUE(searchResultOfFindServiceWithFindHandler.empty());
 }
 
+TEST_F(ServiceDiscovery_test, FindServiceWithEmptyCallableDoesNotDie)
```

Comment:
I would surely hope it does not die but the real semantics we implicitly defined is that it does nothing, even if the search would match something. Feel free to test this, but it is another indicator of a flawed design of `function_ref` or the interface.